src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1366,20 +1367,48 @@ public void AssertMetadataKey(string id, BlittableJsonReaderObject document, Doc
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static Document ParseDocument(JsonOperationContext context, ref TableValueReader tvr, DocumentFields fields)
+        private static Document ParseDocument(JsonOperationContext context, ref TableValueReader tvr, DocumentFields fields, bool skipMetadataData = false)
         {
             if (fields == DocumentFields.All)
             {
+                var id = TableValueToId(context, (int)DocumentsTable.Id, ref tvr);
```

Comment:
Can we do this in only in `StudioCollectionsHandler.cs` and not here?

Reply:
done